File: haskell/Menu.hs
module Menu where

import Control.Concurrent (threadDelay)
import Control.Exception
import Data.Char (isAlphaNum, isDigit, isLower, isUpper)
import Data.List (isInfixOf)
import Data.Typeable (Typeable)
import Utils 
import Inicial 
import Navegacao 
import Missoes 
import Jogo 
import System.Directory (doesFileExist)
import Flashcard (iniciarTreino)

menuPrincipal :: IO ()
menuPrincipal = do
    let largura = larguraTerminal
    let opcoes = [ "🎮 Jogar"
                 , "📰 Ver Regras do Jogo"
                 , "🗺️  Ver Mapa de Missões"
                 , "🎯 Modo Treino"
                 , "🚪 Sair"
                 ]
                          
    escolha <- escolherOpcaoComTitulo "../banners/menu_principal.txt" opcoes
    limparTelaCompleta
    case escolha of
        0 -> do
            putStrLn "Iniciando novo jogo..."
            menuJogo
            menuPrincipal
        1 -> do
            limparTelaCompleta
            mostrarLogoCentralizada "../banners/regras.txt"
            putStrLn "\nPressione Enter para voltar ao menu..."
            _ <- getLine
            menuPrincipal
        2 -> do
            limparTelaCompleta
            imprimirMapa
            putStrLn "\nPressione Enter para voltar ao menu..."
            _ <- getLine
            menuPrincipal
        3 -> do
            putStrLn "Modo Treino"
            iniciarTreino []
            menuPrincipal
            menuPrincipal
        4 -> do
            putStrLn "Saindo do jogo... Até a próxima! 👋"
        _ -> putStrLn "Opção inválida."

File: prolog/menu.pl
:- module(menu, [menu_principal/0]).

:- use_module(utils).
:- use_module(auth).
:- use_module(jogo).
:- use_module(navegacao).
:- use_module(missoes).
:- use_module(treino).

menu_principal :-
    limpar_tela_completa,
    login_corrente(User, _, _, _, _),
    Opcoes = [
        "🎮 Jogar",
        "📰 Ver Regras do Jogo",
        "🗺️ Ver Mapa de Missões",
        "🎯 Modo Treino",
        "🚪 Salvar e Sair"
    ],
    navegacao:escolher_opcao_titulo("../banners/menu_principal.txt", Opcoes, User, Escolha),
    tratar_escolha(Escolha, User).

tratar_escolha(0, User) :-
    jogo:iniciar_selecao_missao(User),
    menu_principal.

tratar_escolha(1, _) :-
    limpar_tela_completa,
    mostrar_banner('../banners/regras.txt'),
    utils:pressionar_enter_voltar,
    menu_principal.

tratar_escolha(2, _) :-
    limpar_tela_completa,
    mostrar_banner('../banners/mapa.txt'),
    utils:pressionar_enter_voltar,
    menu_principal.

tratar_escolha(3, _) :-
    limpar_tela_completa,
    iniciar_treino.

tratar_escolha(4, _) :-
    writeln("Finalizando a sessão..."),
    auth:finalizar_sessao,
    writeln("Até a próxima! 👋"),
    halt.

tratar_escolha(quit, _) :-
    menu_principal.

tratar_escolha(_, _) :-
    writeln("Opção inválida."),
    sleep(1),
    menu_principal.
